sanic/utils.py

```diff
@@ -0,0 +1,69 @@
+from os import environ as os_environ
+from re import findall as re_findall
+from importlib.util import spec_from_file_location, \
+                           module_from_spec
+from typing import Union
+
+
+
+def str_to_bool(val):
+    """Takes string and tries to turn it into bool as human would do.  
+
+    If val is in case insensitive ("y", "yes", "yep", "yup", "t", "true", "on", "enable", "enabled", "1") returns True.  
+    If val is in case insensitive ("n", "no", "f", "false", "off", "disable", "disabled", "0") returns False.  
+    Else Raise ValueError."""
+
+    val = val.lower()
+    if val in ("y", "yes", "yep", "yup", "t", "true", "on", "enable", "enabled", "1"):
```

Comment:
Can we check if value is present in `set`? it will be a slightly faster.

Something like
```
val in {'y', 'es', 'yep', 'yup', 't', 'true', 'on', 'enable', 'enabled', '1'}
```

Reply:
Or maybe ???:  
    if {val}.intersection({'y', 'es', 'yep', 'yup', 't', 'true', 'on', 'enable', 'enabled', '1'}

Reply:
`x in y` is more readable and understandable than `instersection`.

But, I am all for speed. Which of the three options is fastest? If it is `intersection` then just add a comment to explain why that was chosen.

Reply:
The winner is:  

    val in {'y', 'es', 'yep', 'yup', 't', 'true', 'on', 'enable', 'enabled', '1'}

Here is how I measured it:  

    stm1='''
    val in ("y", "yes", "yep", "yup", "t", "true", "on", "enable", "enabled", "1")'''

    stm2='''
    val in {"y", "yes", "yep", "yup", "t", "true", "on", "enable", "enabled", "1"}'''

    stm3='''
    {val}.intersection({"y", "es", "yep", "yup", "t", "true", "on", "enable", "enabled", "1"})'''

    setup1="val=\"on\""

    setup2="val=\"XXX\""

    timeit.repeat(stm1, setup1)
    [0.09617761895060539, 0.09550061076879501, 0.09542451426386833]

    timeit.repeat(stm2, setup1)
    [0.02791474759578705, 0.02763954922556877, 0.02754434570670128]

    timeit.repeat(stm3, setup1)
    [0.5414496809244156, 0.5408613793551922, 0.5403690747916698]

    timeit.repeat(stm1, setup2)
    [0.15312721580266953, 0.15300191193819046, 0.15259921550750732]

    timeit.repeat(stm2, setup2)
    [0.027631547302007675, 0.027267344295978546, 0.027293745428323746]

    timeit.repeat(stm3, setup2)
    [0.5335312411189079, 0.5335069410502911, 0.5330434367060661]